mne/morph.py

```diff
@@ -446,6 +456,55 @@ def apply(self, stc_from, output='stc', mri_resolution=False,
                 output=output)
         return out
 
+    @verbose
+    def compute_vol_morph_mat(self, *, verbose=None):
```

Comment:
the SourceMorph object is meant to be constructed from compute_source_morph and has only 2 public methods (apply and save). I would therefore expect the precomptuation of the vol morph mat to be exposed only via compute_source_morph.

Reply:
Indeed I had it that way originally actually, but this way is actually more functional: it allows people who have already saved volume morphs (from 0.21, etc.) to load them, do this computation, and then save without needing to call `compute_source_morph` will all the same parameters they did the first time.

I also don't see any problem with adding more methods to this object if we need them.

Reply:
(For example I just had 100 subjects I did this with, it motivated me to do it this way)

Reply:
I slept over it and I think we should expose a precompute parameter in compute_source_morph. Users don't have to understand the concepts of morph_mat  which I doubt many users understand. It's now completely hidden in the surface morphing API. So what I would suggest is:

`compute_source_morph(..., precompute='auto')`

where it's True by default for surface and False for volume.

then we can keep the compute_vol_morph_mat method public.

and what would be just great would be to automatically compute the vol_morph_mat on apply if you can infer if it's going to be faster to first precompute and then apply with a dot.

Reply:
> where it's True by default for surface and False for volume.

I'm not going to implement it quite this way because we never want to / cannot turn off precompute for surface -- we actually removed the code path that did it at some point because it wasn't faster unless you had a single time point, and even in that case it wasn't much faster. But I can leave it as `precompute : bool` in case at some point we want to add other options.